supervisor/backups/backup.py

```diff
@@ -725,7 +745,7 @@ def is_excluded_by_filter(item_arcpath: PurePath) -> bool:
 
                 return False
 
-            with self._outer_secure_tarfile.create_inner_tar(
+            with cast(SecureTarFile, self._outer_secure_tarfile).create_inner_tar(
```

Comment:
Is that really necessary? So mypy doesn't remember that this can't be None? :thinking: 

Hm, I guess it could be `None` since the state changed at that point.

Maybe better to store it in a local variable of type `SecureTarFile`, and access that from the inner function?

Reply:
Yea I thought that was odd too. Storing it in another variable after the first check sounds like a good solution.